### PR TITLE
cephadm/services/ingress: fixed keepalived config bug

### PR DESCRIPTION
**Pull request by thegreenbear**
In some special conditions, such as ingress service reconfiguration, cepham would list the network IPs and pick the VIP as the unicast IP of the peer. It would then generate buggy config causing service misbehavior.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

### CODE REVIEW

**Comment by thegreenbear** [comment]
I don't have access to the tracker yet. I will create a bug once my account is approved. And I will edit this PR.

**Review by phlogistonjohn — COMMENTED** [review]
The change seems OK on first glance, but this is the kind of thing that seems to call out for a unit test. Could you please look into adding a unit test to help validate this change?

**Comment by thegreenbear** [comment]
> The change seems OK on first glance, but this is the kind of thing that seems to call out for a unit test. Could you please look into adding a unit test to help validate this change?

Sure. I could try. But I will probably need support. 
First to add a unit test. Second to add a unit test that will mock network interfaces.
I'll try something and then we can discuss on the MR.

**Comment by thegreenbear** [comment]
@phlogistonjohn , WDYT?
Also, is there a way to run a unittest with different configs easily (without duplicating it all)?
I think it could make sense to run the `test_ingress_config` test with different specs (especially with vip network configured or not).

**Comment by thegreenbear** [comment]
@phlogistonjohn , what would be the next step here?

**Review by phlogistonjohn — APPROVED** [review]
So sorry for taking this long to review. I had the PR open in a tab for over a week, meaning to get back to it but never did. But your recent ping has done the trick.

Thanks for adding the test, lgtm.

**Review by adk3798 — APPROVED** [review]

**Comment by thegreenbear** [comment]
> So sorry for taking this long to review. I had the PR open in a tab for over a week, meaning to get back to it but never did. But your recent ping has done the trick.
> 
> Thanks for adding the test, lgtm.

No worry. Cheers!

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-06-28_04:58:18-orch:cephadm-wip-adk-testing-2024-06-27-0902-distro-default-smithi/

Failures all fell under

- cluster log failures (known issue, being cleaned up)
- mds_upgrade_Sequence (known issue)
- staggered upgrade with agent on (known issue)

Nothing to block merging

**Pull request merged**
